src/Engines/MeiliSearchEngine.php

```diff
@@ -201,17 +209,22 @@ public function map(Builder $builder, $results, $model)
             return $model->newCollection();
         }
 
-        $objectIds = collect($results['hits'])->pluck($model->getKeyName())->values()->all();
+        $hits = collect($results['hits'])->keyBy($model->getKeyName());
 
+        $objectIds = $hits->pluck($model->getKeyName())->values()->all();
         $objectIdPositions = array_flip($objectIds);
 
-        return $model->getScoutModelsByIds(
-            $builder, $objectIds
-        )->filter(function ($model) use ($objectIds) {
-            return in_array($model->getScoutKey(), $objectIds);
-        })->sortBy(function ($model) use ($objectIdPositions) {
-            return $objectIdPositions[$model->getScoutKey()];
-        })->values();
+        return $model->getScoutModelsByIds($builder, $objectIds)
+            ->filter(function ($model) use ($objectIds) {
+                return in_array($model->getScoutKey(), $objectIds);
+            })
+            ->each(function ($model) use ($hits) {
+                $this->setHitMetadata($model, $hits);
+            })
+            ->sortBy(function ($model) use ($objectIdPositions) {
+                return $objectIdPositions[$model->getScoutKey()];
+            })
+            ->values();
```

Comment:
Can you undo the style changes you made here so it's more easier to parse the diff?